crates/engine/primitives/src/config.rs

```diff
@@ -17,7 +17,29 @@ pub const DEFAULT_RESERVED_CPU_CORES: usize = 1;
 const DEFAULT_BLOCK_BUFFER_LIMIT: u32 = 256;
 const DEFAULT_MAX_INVALID_HEADER_CACHE_LENGTH: u32 = 256;
 const DEFAULT_MAX_EXECUTE_BLOCK_BATCH_SIZE: usize = 4;
-const DEFAULT_CROSS_BLOCK_CACHE_SIZE: u64 = 4 * 1024 * 1024 * 1024;
+const DEFAULT_CROSS_BLOCK_CACHE_SIZE_PERCENT: f64 = 0.3;
+
+fn get_dynamic_cross_block_cache_size() -> u64 {
+    #[cfg(feature = "std")]
+    {
+        use sysinfo::System;
+
+        let mut sys = System::new_all();
+        sys.refresh_memory();
+
+        let total_memory = sys.total_memory();
+        let used_memory = sys.used_memory();
+        let available_memory = total_memory - used_memory;
```

Comment:
This calculation is redundant. Use https://docs.rs/sysinfo/latest/sysinfo/struct.System.html#method.available_memory